src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -172,7 +170,7 @@ public async Task StopAsync(CancellationToken cancellationToken)
         public void Cancel()
         {
             ThrowIfDisposed();
-            _cancellationTokenSource.Cancel();
+            StopAsync(CancellationToken.None).Wait();

```

Comment:
I think we have the `CancellationToken` for graceful shutdown:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-class-library?tabs=v2%2Ccmd#cancellation-tokens